controllers/metal3.io/baremetalhost_controller.go

```diff
@@ -432,7 +432,10 @@ func (r *BareMetalHostReconciler) actionRegistering(prov provisioner.Provisioner
 	registeredNewCreds := !info.host.Status.GoodCredentials.Match(*info.bmcCredsSecret)
 	info.host.UpdateGoodCredentials(*info.bmcCredsSecret)
 
-	info.host.ClearError()
+	// Avoid to clean up other kind of errors
+	if info.host.Status.ErrorType == metal3v1alpha1.RegistrationError {
```

Comment:
We should probably also always clear the error if `registeredNewCreds` is true. WDYT?

Reply:
That's an interesting point. I assumed that the error was set only [here](https://github.com/metal3-io/baremetal-operator/blob/0fde063573ec41831016fca8ab3ebbb8b51434b5/controllers/metal3.io/baremetalhost_controller.go#L422) during the registration. In case of credentials change the `ValidateManagementAccess` can fail in the update, but in such case the actions gets simply re-enqueued (409) or an actionError is returned, so it seems that such scenario the ClearError() shouldn't be required for the host.

Reply:
Credentials can change at any time. If someone changes the password on the server prior to changing it in the BMH, then I think it's possible to see another kind of error.
It seems reasonable that once we know the creds have changed (and the change worked), we discard any assumptions about what we thought might have been broken before.

Reply:
Ok, so if IIUC changing the creds could be considered a very "strong" kind of event that somehow resets the current situation (forcing anyhow implicitly a re-evaluation).